src/util.rs

```diff
@@ -145,21 +147,21 @@ struct Align16;
 /// let mut x: AlignedArray<[i16; 64 * 64]> = UninitializedAlignedArray();
 /// assert!(x.array.as_ptr() as usize % 16 == 0);
 /// ```
-pub struct AlignedArray<ARRAY>
+pub struct AlignedArray<ARRAY, AlignType = Align16>
```

Comment:
This part might require a little discussion, but the alignment is a property that should change depending on the platform, not on the usage in the code.

Ideally you could (and should) bump the alignment to 32 for now and hide that detail from the rest of the code.

Reply:
I don't think making it platform specific is perfect since some avx2 code will only need 128 bit alignment.